ext/js/comm/anki-connect.js

```diff
@@ -579,6 +590,43 @@ export class AnkiConnect {
         return /** @type {T[]} */ (result);
     }
 
+    /**
+     * @param {unknown} result
+     * @param {number} expectedCount
+     * @param {string} [context]
+     * @returns {({ canAdd: true } | { canAdd: false, error: string })[]}
```

Comment:
Create a TS type for this, as it is sufficiently complex.

Reply:
done